client/cody-ui/src/chat/TranscriptItem.tsx

```diff
@@ -31,71 +31,141 @@ export const TranscriptItem: React.FunctionComponent<
         inProgress: boolean
+        beingEdited: boolean
+        setBeingEdited: (input: boolean) => void
         fileLinkComponent: React.FunctionComponent<FileLinkProps>
+        textAreaComponent?: React.FunctionComponent<ChatUITextAreaProps>
+        EditButtonContainer?: React.FunctionComponent<EditButtonProps>
+        editButtonOnSubmit?: (text: string) => void
+        showEditButton: boolean
         FeedbackButtonsContainer?: React.FunctionComponent<FeedbackButtonsProps>
         feedbackButtonsOnSubmit?: (text: string) => void
         showFeedbackButtons: boolean
     } & TranscriptItemClassNames
 > = ({
     message,
     inProgress,
+    beingEdited,
+    setBeingEdited,
     fileLinkComponent,
     transcriptItemClassName,
     humanTranscriptItemClassName,
     transcriptItemParticipantClassName,
     codeBlocksCopyButtonClassName,
     transcriptActionClassName,
+    textAreaComponent: TextArea,
+    EditButtonContainer,
+    editButtonOnSubmit,
+    showEditButton,
     FeedbackButtonsContainer,
     feedbackButtonsOnSubmit,
     showFeedbackButtons,
-}) => (
-    <div
-        className={classNames(
-            styles.row,
-            transcriptItemClassName,
-            message.speaker === 'human' ? humanTranscriptItemClassName : null
-        )}
-    >
-        <header className={classNames(styles.participant, transcriptItemParticipantClassName)}>
-            <h2 className={styles.participantName}>
-                {message.speaker === 'assistant' ? (
-                    <>
-                        <CodySvg className={styles.participantAvatar} /> Cody
-                    </>
-                ) : (
-                    'Me'
-                )}
-            </h2>
-            {/* display feedback buttons on last assistant message only */}
-            <div className={styles.participantName}>
-                {showFeedbackButtons &&
-                    FeedbackButtonsContainer &&
-                    feedbackButtonsOnSubmit &&
-                    message.speaker === 'assistant' && (
-                        <FeedbackButtonsContainer
-                            className={styles.FeedbackButtonsContainer}
-                            feedbackButtonsOnSubmit={feedbackButtonsOnSubmit}
-                        />
+}) => {
+    const [formInput, setFormInput] = useState<string>(message.displayText ?? '')
+    const inputHandler = useCallback(
+        (inputValue: string): void => {
+            setFormInput(inputValue)
+        },
+        [setFormInput]
+    )
+    const textarea =
+        TextArea && beingEdited && editButtonOnSubmit ? (
+            <TextArea
+                className={classNames(styles.chatInput)}
+                rows={5}
+                value={formInput}
+                autoFocus={true}
+                required={true}
+                onInput={({ target }) => {
+                    const { value } = target as HTMLInputElement
+                    inputHandler(value)
```

Comment:
```suggestion
                    setFormInput(value)
```
nit: this way we can remove the `inputHandler` above

Reply:
Oops nice catch, copied this line-for-line from somewhere else in the codebase when it isn't necessary here 😅